rust/connlib/tunnel/src/gateway.rs

```diff
@@ -229,14 +251,14 @@ impl GatewayState {
         ipv6: Ipv6Addr,
         stun_servers: HashSet<SocketAddr>,
         turn_servers: HashSet<(RelayId, RelaySocket, String, String, String)>,
-        domain: Option<DomainName>,
+        domain: Option<(DomainName, Vec<IpAddr>)>,
         expires_at: Option<DateTime<Utc>>,
         resource: ResourceDescription<ResolvedResourceDescriptionDns>,
         now: Instant,
     ) -> Result<ConnectionAccepted> {
         match (&domain, &resource) {
-            (Some(domain), ResourceDescription::Dns(r)) => {
-                if !crate::dns::is_subdomain(domain, &r.domain) {
+            (Some((domain, addresses)), ResourceDescription::Dns(r)) => {
+                if !crate::dns::is_subdomain(domain, &r.domain) || addresses.is_empty() {
```

Comment:
So this means, a DNS resource that doesn't have any records yet is just invalid? I think we should be careful with failing from the `accept` function because the client has no idea what is happening then and will just time-out, making this rather hard to debug.

At the minimum, `addresses.is_empty` should log a fat warning on the gateway.

Why do we need to fail the connection here just because there are no DNS records?

Reply:
This also won't work with our idea of making things backwards compatible.